### PR TITLE
[PHX-1022] Create RadioButtonDotless component

### PR DESCRIPTION
**Pull request by micahhahn**
# :star2: Adding a new component

## About the component

This component behaves like a `RadioButton`, but has the look and feel of a normal `Button`.  We want to use this for multiple choice quiz questions so that we can separate the act of a student choosing an option and a student submitting their choice.  This should prevent mistakes for touch screen users and be less frustrating overall. 

I more or less copied the a11y behavior from the `RadioButton` component - so it should be at least as accessible as that! 

## :framed_picture: What does this change look like?

image

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with this component in mind
- [x] Component has clear documentation
- [x] Component is in the Component Catalog
    - [x] Component is categorized reasonably (see [Category](https://github.com/NoRedInk/noredink-ui/blob/master/component-catalog-app/Category.elm) for all the currently available categories). The component can be in multiple categories, if appropriate.
    - [x] Component has a representative preview for the Component Catalog cards (bonus points for making it delightful!)
    - [x] Component has a customizable example. Aim for having _every_ possible supported version of the component displayable through the configuration on this page. (Protip: This is handy for testing expected behavior!)
    - [x] The customizable example produces sample code
    - [x] The component's example page includes keyboard behavior, if any
    - [x] The component's example page includes guidance around how to use the component, if necessary
- [x] Component is tested
    - axe checks and percy snapshots will be automatically added for every component. However, if you want to customize _when_ the checks and snapshots are made, you will need to make changes to [script/puppeteer-tests.js](https://github.com/NoRedInk/noredink-ui/blob/master/script/puppeteer-tests.js).
    - there are 2 ways to add Elm tests:
        - if you want to test the Component Catalog example directly, add ProgramTest-style tests under `component-catalog/tests`
        - if you want to test the component directly, add tests under `tests/Spec`. Historically, this has been the more popular Elm testing strategy for noredink-ui.
- [x] Component API follows standard patterns in noredink-ui
    - e.g., as a dev, I can conveniently add an `nriDescription`
    - and adding a new feature to the component will _not_ require major API changes to the component
- [x] Please assign [team-accessibilibats-a11ybats](https://github.com/orgs/NoRedInk/teams/team-accessibilibats-a11ybats) as a reviewer in addition to assigning a reviewer from your team

### CODE REVIEW

**Comment by micahhahn** [comment]
@ravi-morbia Let me know when you're ready to do a design review on this new component to make sure it will match the expectations laid out in [Figma](https://www.figma.com/file/swxmvrKsgawbuicqHtdwN8/Learning-XP-(2023%2B)?node-id=3796%3A99893&mode=dev)

I'll serve up my local dev environment so you can poke around with the layouts in the component catalog

File: src/Nri/Ui/FocusRing/V1.elm [hunk 1]
```diff
@@ -53,9 +53,6 @@ forMouseUsers =
     , Css.Global.selector ":focus-within .checkbox-icon-container"
         [ Css.important (Css.boxShadow Css.none)
         ]
-    , Css.Global.selector ":focus-within .Nri-RadioButton-RadioButtonIcon"
-        [ Css.important (Css.boxShadow Css.none)
-        ]
```
Comment (micahhahn) [inline src/Nri/Ui/FocusRing/V1.elm:58]: @tesk9 I thought really hard about if there was a good reason to have this selector here twice! Couldn't come up with any, pulled it, and it still works! 

Please let me know if this was indeed intentional!

**Comment by ravi-morbia** [comment]
> @ravi-morbia Let me know when you're ready to do a design review on this new component to make sure it will match the expectations laid out in [Figma](https://www.figma.com/file/swxmvrKsgawbuicqHtdwN8/Learning-XP-(2023%2B)?node-id=3796%3A99893&mode=dev)
> 
> ~I'll serve up my local dev environment again over Tailscale so you can poke around with the layouts in the component catalog~
> 
> ^ kristine noted that you can click "Details" next to the `netlify/noredink-ui/deploy-preview` CI action below to see a preview for yourself! How neat

Hey @micahhahn this is looking good! For multiple choice, I think we would want the same height as large buttons. Maybe to keep this component flexible, would it be possible to do something similar to that, like min-height 35px for small, 45px for medium, and 55px for large or something?

**Comment by micahhahn** [comment]
> For multiple choice, I think we would want the same height as large buttons. Maybe to keep this component flexible, would it be possible to do something similar to that, like min-height 35px for small, 45px for medium, and 55px for large or something?

Definitely! I'll add an option for sizing that exactly mimics button's `Small` / `Medium` / `Large` sizes which are `36px` / `45px` / and `56px` respectively.

**Comment by micahhahn** [comment]
(In general I tried to keep the API as small as possible to meet our *current* needs, but open to extension - if someone else needs features added they should be able to do so without introducing a new version even).

File: src/Nri/Ui/FocusRing/V1.elm (hunk 1 above)
Comment (tesk9) [reply src/Nri/Ui/FocusRing/V1.elm:58]: Oh nice catch! Just a mistake, thanks for fixing!

**Comment by micahhahn** [comment]
@ravi-morbia @tesk9 This is ready for a re-review! Size attributes have been added along with some tweaks to behave more like the `Button` component.

**Comment by ravi-morbia** [comment]
Thanks @micahhahn! While playing around with the settings, I saw that an individual word can wrap. Is that supposed to happen? I think we would only want whole words to wrap.

Screenshot 2023-08-21 at 9 46 01 AM

File: src/Nri/Ui/RadioButtonDotless/V1.elm [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+            (valueToString value)
+            isChecked
+            ([ Attributes.id idValue
+             , Attributes.class "Nri-RadioButton-HiddenRadioInput"
+             , case config.onSelect of
+                Just onSelect_ ->
+                    Events.onClick (onSelect_ value)
+
+                Nothing ->
+                    AttributeExtra.none
```
Comment (tesk9) [inline src/Nri/Ui/RadioButtonDotless/V1.elm:267]: I know this isn't something that `RadioButton` does particularly well, but I wonder if we could disable the `RadioButtonDotless` when no `onSelect` is present?

If not, maybe in the `preview`, we could pass through custom attributes of `tabindex=0` and `aria-disabled=true`... 

Right now, these new radio buttons are tabstops on the "catalog" view, is all.

**Review by tesk9 — APPROVED** [review]
Looking fabulous, thank you!

**Comment by micahhahn** [comment]
> Thanks @micahhahn! While playing around with the settings, I saw that an individual word can wrap. Is that supposed to happen? I think we would only want whole words to wrap.

Hey @ravi, I copied that from the Radio Button code... [this is the PR that introduced the changes to radio buttons](https://github.com/NoRedInk/noredink-ui/pull/1296).  I guess we had issues with long email addresses being used as radio buttons? 

I think we're fine to drop back to normal behavior and we can change it later if need be.

**Comment by ravi-morbia** [comment]
Ah gotcha, okay sounds good @micahhahn!

**Pull request merged**